src/exchanges/ZeroExV3Adapter.sol

```diff
@@ -0,0 +1,466 @@
+    function approveAssetsMakeOrder(address _targetExchange, IZeroExV3.Order memory _order)
+        internal
+    {
+        approveAsset(
+            getAssetAddress(_order.makerAssetData),
+            getAssetProxy(_targetExchange, _order.makerAssetData),
+            _order.makerAssetAmount,
+            "makerAsset"
+        );
+        if (_order.makerFee > 0) {
+            approveAsset(
+                getAssetAddress(_order.makerFeeAssetData),
+                getAssetProxy(_targetExchange, _order.makerFeeAssetData),
+                _order.makerFee,
+                "makerFeeAsset"
+            );
+        }
+    }
+
+    // @notice Approves takerAsset, takerFeeAsset, protocolFee
+    function approveAssetsTakeOrder(address _targetExchange, IZeroExV3.Order memory _order)
+        internal
+    {
+        approveProtocolFeeAsset(_targetExchange);
+        approveAsset(
+            getAssetAddress(_order.takerAssetData),
+            getAssetProxy(_targetExchange, _order.takerAssetData),
+            _order.takerAssetAmount,
+            "takerAsset"
+        );
+        if (_order.takerFee > 0) {
+            approveAsset(
+                getAssetAddress(_order.takerFeeAssetData),
+                getAssetProxy(_targetExchange, _order.takerFeeAssetData),
+                _order.takerFee,
+                "takerFeeAsset"
+            );
+        }
+    }
+
+    function approveProtocolFeeAsset(address _targetExchange) internal {
+        address protocolFeeCollector = IZeroExV3(_targetExchange).protocolFeeCollector();
+        uint256 protocolFeeAmount = calcProtocolFeeAmount(_targetExchange);
+        if (protocolFeeCollector == address(0) || protocolFeeAmount == 0) return;
+
+        Hub hub = getHub();
+        address nativeAsset = Accounting(hub.accounting()).NATIVE_ASSET();
+
+        approveAsset(nativeAsset, protocolFeeCollector, protocolFeeAmount, "protocolFee");
+    }
+
+    function executeFill(
+        address _targetExchange,
+        IZeroExV3.Order memory _order,
+        uint256 _takerAssetFillAmount,
+        bytes memory _signature
+    )
+        internal
+        returns (uint256)
+    {
+        Hub hub = getHub();
+        address makerAsset = getAssetAddress(_order.makerAssetData);
+        uint preMakerAssetBalance = IERC20(makerAsset).balanceOf(address(this));
+
+        IZeroExV3.FillResults memory fillResults = IZeroExV3(_targetExchange).fillOrder(
+            _order,
+            _takerAssetFillAmount,
+            _signature
+        );
+
+        uint256 postMakerAssetBalance = IERC20(makerAsset).balanceOf(address(this));
+
+        // Account for case where makerAsset, takerFee, protocolFee are the same
+        uint256 makerAssetFeesTotal;
+        if (
+            makerAsset == Accounting(hub.accounting()).NATIVE_ASSET() &&
+            IZeroExV3(_targetExchange).protocolFeeCollector() != address(0)
+        )
+        {
+            makerAssetFeesTotal += calcProtocolFeeAmount(_targetExchange);
```

Comment:
Changed to assign `makerAssetFeesTotal = calcProtocolFeeAmount(_targetExchange)` since it's the first operation after defining the var with a default value of 0.